x/concentrated-liquidity/pool_test.go

```diff
@@ -12,50 +12,182 @@ import (
+	s.Require().Equal(expectedTokenInDelta.String(), tokenInDelta.String())
+	s.Require().Equal(expectedTokenOutDelta.String(), tokenOutDelta.String())
+
+	// check the new tick is at the expected value
+	s.Require().Equal(sdk.NewInt(85180).String(), updatedTick.String())
+
+	// check pool liquidity
+	lowerSqrtPrice, err := s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(lowerTick)
+	s.Require().NoError(err)
+	upperSqrtPrice, err := s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(upperTick)
+	s.Require().NoError(err)
+	expectedLiquidity := cl.GetLiquidityFromAmounts(currSqrtPrice, lowerSqrtPrice, upperSqrtPrice, amount0.Mul(sdk.NewInt(2)), amount1.Mul(sdk.NewInt(2)))
+	s.Require().Equal(expectedLiquidity.String(), updatedLiquidity.String())
+
+	//
+	// TEST 2: one price range
+	//
+
+	// we use the same price range as above, but just with a single position instead of two
+
+	// 1 eth 5000 usdc position
+	amount0 = sdk.NewInt(1000000)
+	amount1 = sdk.NewInt(5000000000)
+
+	// create pool
+	pool, err = s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(ctx, 2, "eth", "usdc", currSqrtPrice, currTick)
+	s.Require().NoError(err)
+
+	// add position
+	_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, pool.Id, s.TestAccs[1], amount0, amount1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+	s.Require().NoError(err)
+
+	// swapping parameters used for test
+	// swap in 42 usdc for some amount of eth
+	tokenIn = sdk.NewCoin("usdc", sdk.NewInt(42000000))
+	tokenOutDenom = "eth"
+	// set no swap fee
+	swapFee = sdk.ZeroDec()
+	// limit max price impact to 5004 usdc per eth
+	priceLimit = sdk.NewDec(5004)
+
+	// run calculation
+	tokenInDelta, tokenOutDelta, updatedTick, updatedLiquidity, err = s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, priceLimit, pool.Id)
+	s.Require().NoError(err)
+
+	// we expect to put 41999999 usdc in and in return get .008396 eth back
+	expectedTokenInDelta = sdk.NewInt(41999999)
+	expectedTokenOutDelta = sdk.NewInt(8396)
+
+	// ensure tokenIn and tokenOut meet our expected values
+	s.Require().Equal(expectedTokenInDelta.String(), tokenInDelta.String())
+	s.Require().Equal(expectedTokenOutDelta.String(), tokenOutDelta.String())
+
+	// this is off by one (too large), I think it is the priceToTick func, try using ln PR from main
+	s.Require().Equal(sdk.NewInt(85184).String(), updatedTick.String())
+
+	// check pool liquidity
+	lowerSqrtPrice, err = s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(lowerTick)
+	s.Require().NoError(err)
+	upperSqrtPrice, err = s.App.ConcentratedLiquidityKeeper.TickToSqrtPrice(upperTick)
+	s.Require().NoError(err)
+	expectedLiquidity = cl.GetLiquidityFromAmounts(currSqrtPrice, lowerSqrtPrice, upperSqrtPrice, amount0, amount1)
+	s.Require().Equal(expectedLiquidity.String(), updatedLiquidity.String())
+
+	//
+	// TEST 3: two consecutive price ranges
```

Comment:
Consider separating these tests out

Reply:
can you explain what you mean by separating the tests out?

Reply:
Nm, I will create a PR for making this TDD